### PR TITLE
fix: only cleanup allocated stub drives

### PR DESCRIPTION
**Pull request by austinvazquez**
*Issue #, if available:*
#630

*Description of changes:*
Only cleanup stub drives for tasks that have allocated them.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Signed-off-by: Austin Vazquez <macedonv@amazon.com>

### CODE REVIEW

File: runtime/service.go [hunk 1]
```diff
@@ -1281,7 +1281,7 @@ func (s *service) Delete(requestCtx context.Context, req *taskAPI.DeleteRequest)
 	var result *multierror.Error
 	// Trying to release stub drive for further reuse
 	err = s.containerStubHandler.Release(requestCtx, req.ID, s.driveMountClient, s.machine)
-	if err != nil {
+	if err != nil && !errors.Is(err, ErrDriveNotFound) {
 		result = multierror.Append(result, errors.Wrapf(err, "failed to release stub drive for container: %s", req.ID))
```
Comment (kzys) [inline runtime/service.go:1285]: Hmm, why does it release drives that are not there?

Comment (kzys) [reply runtime/service.go:1285]: And Delete() releases all of them regardless of they are managed or not?

Is it possible to handle that inside the implementation of Release()? (cc @Kern--)

Comment (austinvazquez) [reply runtime/service.go:1285]: The problem is on task creation we know the container is mounted inside the VM by inspecting the `rootfs` on the request. On deletion however, all we have is the request ID. Another solution is to implement a stub handler function to check if the drive is in use before attempting to release, but that feels more weak than a no-op.

Comment (Kern--) [reply runtime/service.go:1285]: I think we probably want to do the book keeping in the service to know whether the container is drive-backed or inside the VM. It's pretty easy for implicit fallback to either mask bugs later, or make debugging more difficult, or just make it harder to understand the code from an outside perspective.

Comment (austinvazquez) [reply runtime/service.go:1285]: Okay, I'll try my hand at that solution and get another revision out.

File: runtime/service.go [hunk 2]
```diff
@@ -143,6 +143,8 @@ type service struct {
 	driveMountStubs          []MountableStubDrive
 	exitAfterAllTasksDeleted bool // exit the VM and shim when all tasks are deleted
 
+	taskCleanupStack map[string][]func(context.Context) error
```
Comment (kzys) [inline runtime/service.go:146]: Let's keep it simple. We right now don't need an arbitrary function and one task wouldn't need multiple cleanup functions.

**Review by kzys — APPROVED** [review]

**Pull request merged**